src/utils/safe-messages.ts

```diff
@@ -16,12 +22,22 @@ export const generateSafeMessageMessage = (message: SafeMessage['message']): str
  * @param message Message to sign
  * @returns `SafeMessage` types for signing
  */
-export const generateSafeMessageTypedData = (safe: SafeInfo, message: SafeMessage['message']): EIP712TypedData => {
+export const generateSafeMessageTypedData = (
+  { version, chainId, address }: SafeInfo,
+  message: SafeMessage['message'],
+): EIP712TypedData => {
+  if (!version) {
+    throw Error('Cannot create SafeMessage without version information')
+  }
+  const isHandledByFallbackHandler = gte(version, EIP1271_FALLBACK_HANDLER_SUPPORTED_SAFE_VERSION)
+
   return {
-    domain: {
-      chainId: safe.chainId,
-      verifyingContract: safe.address.value,
-    },
+    domain: isHandledByFallbackHandler
+      ? {
+          chainId: chainId,
+          verifyingContract: address.value,
+        }
+      : { verifyingContract: address.value },
```

Comment:
could we pass
```suggestion
   domain: {
      chainId: isHandledByFallbackHandler ? chainId : undefined,
      verifyingContract: address.value,
    },
```

Reply:
I'm not sure if an undefined value will have impact on the resulting hash. I will give it a try.

Reply:
`undefined` values will cause errors when creating the hash with ethers. So we cannot rewrite it.